src/main/java/com/iota/iri/service/ledger/impl/LedgerServiceImpl.java

```diff
@@ -173,11 +186,15 @@ public boolean isBalanceDiffConsistent(Set<Hash> approvedHashes, Map<Hash, Long>
                             if (transactionViewModel.getCurrentIndex() == 0) {
                                 boolean validBundle = false;
 
-                                final List<List<TransactionViewModel>> bundleTransactions = BundleValidator.validate(
+                                final List<List<TransactionViewModel>> bundleTransactions = bundleValidator.validate(
                                         tangle, snapshotProvider.getInitialSnapshot(), transactionViewModel.getHash());
 
                                 for (final List<TransactionViewModel> bundleTransactionViewModels : bundleTransactions) {
 
+                                    //ISSUE 1008: generateBalanceDiff should be refactored so we don't have those hidden
+                                    // concerns
+                                    persistValidatedSpentAddresses(bundleTransactionViewModels);
```

Comment:
this appears before the `BundleValidator.isInconsistent(bundleTransactionViewModels), etc.` - it seems you are adding an address even if the signature is invalid/has no funds? (I'm looking in GH, not an IDE so I might be mistaken)

Reply:
It is indeed before `BundleValidator.isInconsistent(bundleTransactionViewModels)`. However, it is after `bundleValidator.validate()` which does signature validation. So I believe it is the correct place to put the method.